documentation/modules/security/proc-replacing-your-own-private-keys.adoc

```diff
@@ -0,0 +1,136 @@
++
+Following an example of the `Secret` related to the cluster CA certificate for a Kafka cluster named `my-cluster`.
++
+[source,yaml,subs=attributes+]
+----
+apiVersion: v1
+kind: Secret
+data:
+  ca.crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0F... <1>
+metadata:
+  annotations:
+    strimzi.io/ca-cert-generation: "0" <2>
+  labels:
+    strimzi.io/cluster: my-cluster
+    strimzi.io/kind: Kafka
+  name: my-cluster-cluster-ca-cert
+  #...
+type: Opaque
+----
+<1> Current CA certificate base64 encoded
+<2> Current CA certificate generation
+
+.. Retain the current CA certificate.
++
+Rename the current `ca.crt` field in the `data` section as `ca-__DATE__.crt`, where _DATE_ is the certificate expiry date in the format _YEAR-MONTH-DAYTHOUR-MINUTE-SECONDZ_.
+For example `ca-2018-09-27T17-32-00Z.crt`.
+Leave the value for the field as it is to retain the current CA certificate.
+
+.. Encode your new CA certificate into base64.
++
+[source,shell,subs="+quotes"]
+cat _PATH-TO-NEW-CERTIFICATE_ | base64
+
+.. Update the CA certificate.
++
+Create a new `ca.crt` field in the `data` section of the `Secret` and copy the new CA certificate base64 encoded got from the previous step
++
+.. Increase the CA certificate generation.
++
+Updated the `strimzi.io/ca-cert-generation` annotation with a higher incremental value.
+For example, change `strimzi.io/ca-cert-generation=0` to `strimzi.io/ca-cert-generation=1`.
++
+
+.. Save changes and exit.
++
+Save the changes to the `Secret` and exit.
+
+. Update the corresponding `Secret` with the new CA key and generation.
++
+.. Edit the existing `Secret`:
++
+[source,shell,subs="+quotes"]
+kubectl edit secret _CA-KEY-SECRET_
++
+_CA-KEY-SECRET_ is the name of CA key, which is `_KAFKA-CLUSTER-NAME_-cluster-ca` for the cluster CA key and `_KAFKA-CLUSTER-NAME_-clients-ca` for the clients CA key.
++
+Following an example of the `Secret` related to the cluster key CA for a Kafka cluster named `my-cluster`.
++
+[source,yaml,subs=attributes+]
+----
+apiVersion: v1
+kind: Secret
+data:
+  ca.key: LS0tLS1CRUdJTiBDRVJUSUZJQ0F... <1>
+metadata:
+  annotations:
+    strimzi.io/ca-key-generation: "0" <2>
+  labels:
+    strimzi.io/cluster: my-cluster
+    strimzi.io/kind: Kafka
+  name: my-cluster-cluster-ca
+  #...
+type: Opaque
+----
+<1> Current CA key base64 encoded
+<2> Current CA key generation
+
+.. Encode the new key used to sign the new CA certificate into base64.
++
+[source,shell,subs="+quotes"]
+cat _PATH-TO-NEW-KEY_ | base64
+
+.. Update the CA key.
++
+Copy the new CA key base64 encoded got from the first step into the `ca.key` field of the `data` section in the `Secret`.
++
+.. Increase the CA key generation.
++
+Updated the `strimzi.io/ca-key-generation` annotation with a higher incremental value.
+For example, change `strimzi.io/ca-key-generation=0` to `strimzi.io/ca-key-generation=1`.
++
+
+. Save changes and exit.
```

Comment:
I guess this can still screw up the cluster if only one secret is updated when a recocniliation starts? This process is f*cked up :-/. Should we tell the users to pause the reconciliation or something?

Reply:
Good point. Yeah it could happen that CA cert is updated but not corresponding key. During rolls it will screw up stuff. 
I think that pausing and resuming is the only good solution we have. I will add a step at the beginning and the end of the procedure.